dissect/target/helpers/regutil.py

```diff
@@ -28,6 +29,33 @@
 """The possible value types that can be returned from the registry."""
 
 
+def lazy(postponed: Callable) -> Callable:
+    def decorator(func: Callable) -> Callable:
+        def inner(self: Any) -> Any:
+            flag = f"_{postponed.__name__}_called"
+            if not hasattr(self, flag):
+                postponed(self)
+                setattr(self, flag, True)
+            return func(self)
+
+        return inner
+
+    return decorator
+
+
```

Comment:
```suggestion
```

Reply:
I don't really see how cached_property can implement this, since you have a caching/lazy-issue on object level, not property level, by referring to a method you can attach multiple lazy properties (i.e lazy != cached). However, if you do not like having a custom decorator for this, I can just revert it to the existing code.

Reply:
Can you elaborate further, maybe with an example? I'm not following. I did write the `@cached_property` example without testing, but I fail to see how it doesn't solve it?

Reply:
There are 2 properties that need to be calculated (type and value). Upon accessing value they will both be calculated but only value will be cached. If you then access type the logic will rerun because type has not been cached yet. Or you need a way to prefill the cache of another property, but that also feels a bit odd. Or maybe I do not understand the full functionality of cached_property or I miss something else. But re-parsing twice because of a missed cache seems inefficient to me? This is why I say the laziness is on the object level (since two separate properties need to be cached in one go).

Reply:
Did you try this hypothesis on the code I suggested?

Reply:
Sorry I was misreading your suggestion, I see you moved the caching to the calculating function and derive the values in the respective properties. That also works.